src/parser/jobs/brd/PitchPerfect.js

```diff
@@ -0,0 +1,253 @@
+					}
+					// AND ALSO FOR RANGED TRAIT, BECAUSE APPARENTLY IT'S PHYSICAL DAMAGE ONLY REEEEEEEEEE
+					fixedMultiplier = Math.trunc((fixedMultiplier + TRAIT_STRENGTH) * 100) / 100
+				}
+
+				// If it's a Pitch Perfect damage
+				if (event.ability.guid === ACTIONS.PITCH_PERFECT.id) {
+
+					ppEvents.push({event: event, rawDamage: event.amount / fixedMultiplier})
+
+				// Otherwise, if it doesn't have a conditional potency (Sidewinder and Pitch Perfect), it will be used to calculate 'K'
+				} else if (event.ability.guid !== ACTIONS.SIDEWINDER.id) {
+
+					// ...let's not count Spears for now
+					if (!this._player.hasStatus(STATUSES.THE_SPEAR.id)) {
+
+						const critTier = this._parseCritBuffs(event)
+
+						if (!damageInstances[critTier]) {
+							damageInstances[critTier] = []
+						}
+
+						damageInstances[critTier].push({event: event, rawDamage: event.amount / fixedMultiplier})
+					}
+				}
+			}
+		}
+
+		// We use the damage events to determine 'K'
+		// tl;dr: 'K' is an approximation to damage to potency ratio, ignoring the natural 5% spread because we don't need this kind of precision
+		const k = this._getK(damageInstances)
+
+		// We now use 'K' to guesstimate PP potency:
+		for (const pp of ppEvents) {
+
+			// We already have the unbuffed damage, we now need to strip PP off crit/dhit mods:
+			let rawDamage = pp.rawDamage
+
+			if (pp.event.criticalHit) {
+				rawDamage = Math.trunc(rawDamage / this._getCritMod(damageInstances))
+			}
+
+			if (pp.event.directHit) {
+				rawDamage = Math.trunc(rawDamage / DHIT_MOD)
+			}
+
+			// We get the approximated potency and then match to the closest real potency
+			const approximatedPotency = rawDamage * 100 / k
+			const potency = matchClosest(ACTIONS.PITCH_PERFECT.potency, approximatedPotency)
+
+			// We then add the amount of stacks to the event
+			pp.event.stacks = ACTIONS.PITCH_PERFECT.potency.indexOf(potency) + 1
+
+		}
+
+		// Return all this shit
+		return events
+	}
+
+	// Returns the enemy statuses state given the ID
+	_getEnemy(targetId) {
+		if (!this._enemies[targetId]) {
+			this._enemies[targetId] = {
+				statuses: {},
+				hasStatus(statusId) { return this.statuses[statusId] || false },
+			}
+		}
+
+		return this._enemies[targetId]
+	}
+
+	// Returns the accumulated crit modifier from all the currently active crit buffs/debuffs
+	_parseCritBuffs(event) {
+		// We need to get the specific enemy in case it's Chain Stratagem
+		const enemy = this._getEnemy(event.targetID)
+		let accumulatedCritBuffs = 0
+
+		for (const modifier of CRIT_MODIFIERS) {
+
+			let hasStatus = false
+
+			if (modifier.id === STATUSES.CHAIN_STRATAGEM.id) {
+				hasStatus = enemy.hasStatus(modifier.id)
+			} else {
+				hasStatus = this._player.hasStatus(modifier.id)
+			}
+
+			if (hasStatus) {
+				accumulatedCritBuffs += modifier.strenght
+			}
+
+		}
+		return accumulatedCritBuffs
+	}
+
+	// Sorry, but these constants are all fucking magic
```

Comment:
There's quite a few repeated numbers in there. Just a few named consts or even just comments would go a long way - as it is, nobody other than yourself is ever gonna be able to update that without spending way too long trying to work out how it does its shit.